src/build_queue.rs

```diff
@@ -726,9 +729,75 @@ pub async fn queue_rebuilds(
     Ok(())
 }
 
+/// Queue rebuilds for failed crates due to a faulty version of rustdoc
+///
+/// It is assumed that the version of rustdoc matches the one of rustc, which is persisted in the DB.
+/// The priority of the resulting rebuild requests will be lower than previously failed builds.
+/// If a crate is already queued to be rebuilt, it will not be requeued.
+/// Start date is inclusive, end date is exclusive.
+#[instrument(skip_all)]
+pub async fn queue_rebuilds_faulty_rustdoc(
+    conn: &mut sqlx::PgConnection,
+    build_queue: &AsyncBuildQueue,
+    start_nightly_date: &NaiveDate,
+    end_nightly_date: &Option<NaiveDate>,
+) -> Result<i32> {
+    let end_nightly_date =
+        end_nightly_date.unwrap_or_else(|| start_nightly_date.succ_opt().unwrap());
+    let mut results = sqlx::query!(
+        r#"
+SELECT c.name,
+       r.version AS "version: Version"
+FROM crates AS c
+         JOIN releases AS r
+              ON c.latest_version_id = r.id
+                  AND r.rustdoc_status = TRUE
+         JOIN LATERAL (
+    SELECT b.id,
+           b.build_status,
+           b.rustc_nightly_date,
+           COALESCE(b.build_finished, b.build_started) AS last_build_attempt
+    FROM builds AS b
+    WHERE b.rid = r.id
+    ORDER BY last_build_attempt DESC
+    LIMIT 1
+    ) AS b ON b.build_status = 'failure' AND b.rustc_nightly_date >= $1 AND b.rustc_nightly_date < $2
+
+"#, start_nightly_date, end_nightly_date
+    )
+        .fetch(&mut *conn);
+
+    let mut results_count = 0;
+    while let Some(row) = results.next().await {
+        let row = row?;
+
+        if !build_queue
+            .has_build_queued(&row.name, &row.version)
+            .await?
+        {
+            results_count += 1;
+            info!(
+                "queueing rebuild for {} {} (priority {})...",
+                &row.name, &row.version, BROKEN_RUSTDOC_REBUILD_PRIORITY
+            );
```

Comment:
this is `tracing::info`, not `log::info!`, where I typically try to use real tracing fields instead of string formatting: 


```
info!(
    name=%row.name,
    version=%row.version,
    priority=BROKEN_RUSTDOC_REBUILD_PRIORITY,
   "queueing rebuild"
```